routes/nodebb.js

```diff
@@ -159,38 +160,59 @@ app.get(`${BASE_REPORT_URL}/user/uid/:uid`, proxyObject());
 
 function isEditablePost() {
   logger.info({message: "isEditablePost method called"});
-  return function(req, res, next) {
+  return async function(req, res, next) {
     logger.info(req.body);
     const uid = parseInt(req.body.uid || req.query.uid, 10);
     const pid = parseInt(req.params.pid, 10);
-    const url = `${req.protocol}://${req.get('host')}${BASE_REPORT_URL}/post/pid/${pid}`
-    const options = {
+    const url = `${req.protocol}://${req.get('host')}${BASE_REPORT_URL}/v3/posts/${pid}`
```

Comment:
Change const to let & use this as base path 
let baseUrl = `${req.protocol}://${req.get('host')}${BASE_REPORT_URL}

let options = {
 method: 'GET',
      json: true
}

try {
options.url: baseUrl + `/v3/posts/${pid}`
...
} catch {
options.url = baseUrl + `/post/pid/${pid}`
...
}

Reply:
Print logger for both the API's. So that we know which api is being called.

Reply:
ok, I will update it.